### ISSUE TITLE
Blackhole / Loopback...

### ISSUE THREAD

**Issue by InThyWord** [ExistentialAudio/BlackHole#187]
Your help would be great to help me truly understand.

I believe I read Blackhole can do the same things as paid product "Loopback" for Macintosh.

I attached a diagram to show what I am looking for.

Can blackhole allow you to direct audio from one place and another microphone source to the same channel as one output/input?

Can blackhole do this? If not, are there any free options which can, or is loopback essential?
<img width="1161" alt="Screenshot 2020-07-28 at 09 25 21" src="https://user-images.githubusercontent.com/68903126/88659558-27a1a700-d0cd-11ea-8359-d072bd38cd20.png">

**Comment by julian-weinert** [comment]
No, BlackHole cannot do this.
It is simply an audio device that can take input and be an input to something else itself.

You need software just like Loopback, except that loopback also allows you to create virtual audio devices like BH.
One could think of loopback to be able to create "BlackHoles".

SoundFlower was something like BH but it is not developed any further. It included SoundFlowerBed which was a basic routing application. You could try finding this, but I'd advice agains using the SoundFlower drivers themselves and the routing application doesn't work to well these days either.

Rogue Amoeba did have a product called "LineIn" which does nothing more. Connecting an input to an output. You can find it in their websites "legacy software" section. It works pretty well, I do use it on a daily basis.

Any other DAW would be fine as well. Or even something like "HostingAU", if you also want to use audio plugins or generators.

**Comment by InThyWord** [comment]
I looked into some of those options. I found that Audio Hijack may in fact do enough without Loopback.

But yes, line-in seemed like it did do it but was not compatible.

Regards.

**Comment by n8gray** [comment]
I just want to mention that the LineIn app can't be used in Catalina. I did get HostingAU to work, which is a decent workaround until somebody puts together a "BlackHoleFlowerBed" routing app. Here's an example of how I use HostingAU to route audio from BH to my Dell monitor:

![Screen Shot 2020-08-28 at 3 25 46 PM](https://user-images.githubusercontent.com/82794/91619810-012c9100-e943-11ea-9cd5-11241a66def2.png)

Then I set the system audio output to BlackHole and voila, the volume slider works for my monitor's speakers!

**Comment by julian-weinert** [comment]
Sad to hear LineIn doesn't work anymore... But perfect, that's the way to go.
I startet something a while ago but had issues to get the rendering working, since I don't have a lot of experience with audio buffers.

**Comment by n8gray** [comment]
My use case is that I'm feeding my laptop's audio to some speakers through my monitor over DisplayPort. Whenever my monitor is connected, I want audio to go to those speakers. MacOS can do this natively, except the monitor's volume controls don't work over DisplayPort, so there's no way to adjust the volume.  By setting up a chain of `System Audio -> BlackHole -> Monitor` I can get working volume control back.

Sadly, the HostingAU solution is better than nothing but it's far from perfect for this use case. It requires a lot of fiddling if you disconnect/reconnect the laptop from the monitor. I'm looking at the CAPlayThrough sample code to see if I can make something better: https://developer.apple.com/library/archive/samplecode/CAPlayThrough/Introduction/Intro.html

**Comment by julian-weinert** [comment]
I've found another solution totally by accident.
It is meant as streamer / broadcasting software, but a 4x2 channel input output mixer.
There are four inputs (one of them BlackHole) and four outputs of which one can be your system output, HDMI or what not.

The cool thing is that it can mix all inputs to all outputs in an extremely simple UI. The streaming aspect can simply be ignored. It is called [LadioCast and is available on the App Store](https://apps.apple.com/de/app/ladiocast/id411213048) for free.

I just discovered it and hope it helps you guys
